thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java

```diff
@@ -201,20 +214,165 @@ public Boolean cloneAnomalyInstances(Long srcId, Long destId) {
     LOG.info("clone merged anomaly results from source anomaly function id {} to id {}", srcId, destId);
 
     List<MergedAnomalyResultDTO> mergedAnomalyResultDTOs = mergedAnomalyResultDAO.findByFunctionId(srcId);
-    if (mergedAnomalyResultDTOs == null) {
+    if (mergedAnomalyResultDTOs == null || mergedAnomalyResultDTOs.isEmpty()) {
       LOG.error("No merged anomaly results found for anomaly function Id: {}", srcId);
       return false;
     }
 
+    LOG.info("{} merged anomaly results has found with functionId {}. Prepare to copy to dest. functionId {}",
+        mergedAnomalyResultDTOs.size(), srcId, destId);
+
+    // TODO: MergedAnomalyResults aren't copied to the new destination
     for (MergedAnomalyResultDTO mergedAnomalyResultDTO : mergedAnomalyResultDTOs) {
       long oldId = mergedAnomalyResultDTO.getId();
       mergedAnomalyResultDTO.setId(null);  // clean the Id, then will create a new Id when save
       mergedAnomalyResultDTO.setRawAnomalyIdList(null);
       mergedAnomalyResultDTO.setFunctionId(destId);
+      mergedAnomalyResultDTO.setFunction(destAnomalyFunction);
       long newId = mergedAnomalyResultDAO.save(mergedAnomalyResultDTO);
       LOG.debug("clone merged anomaly {} to {}", oldId, newId);
     }
     return true;
   }
 
-}
\ No newline at end of file
+  /**
+   * Delete raw or merged anomalies whose start time is located in the given time ranges, except
+   * the following two cases:
+   *
+   * 1. If a raw anomaly belongs to a merged anomaly whose start time is not located in the given
+   * time ranges, then the raw anomaly will not be deleted.
+   *
+   * 2. If a raw anomaly belongs to a merged anomaly whose start time is located in the given
+   * time ranges, then it is deleted regardless its start time.
+   *
+   * If monitoringWindowStartTime is not given, then start time is set to 0.
+   * If monitoringWindowEndTime is not given, then end time is set to Long.MAX_VALUE.
+   * @param monitoringWindowStartTime The start time of the monitoring window (in milli-second)
+   * @param monitoringWindowEndTime The start time of the monitoring window (in milli-second)
+   */
+  @POST
+  @Path("function/{id}/deleteExistingAnomalies")
```

Comment:
This method looks exactly like the cleanup method in the CleanupAndRegenerateAnomaliesTool. Please have a look at CleanupAndRegenerateAnomaliesTool, and remove the duplicate code frmo there. Make the tool call these methods. This file can be the source of the functionality, and the Tool can be another way of accessing them.